src/fs.rs

```diff
@@ -112,6 +114,31 @@ pub fn iter_python_files<'a>(
         })
 }
 
+/// Create tree set with codes matching the pattern/code pairs.
+///
+/// Guarantees a `Some` result will be non-empty
+pub fn ignores_from_path(
+    path: &Path,
+    pattern_code_pairs: &[PerFileIgnore],
+) -> Result<Option<BTreeSet<CheckCode>>> {
+    let (file_path, file_basename) = extract_path_names(path)?;
+    let code_set = {
+        let mut rv = BTreeSet::new();
+        pattern_code_pairs.iter().for_each(|x| {
+            if is_excluded(file_path, file_basename, &[x.pattern.clone()]) {
```

Comment:
Can we try changing `is_excluded` to take `exclude: &[&FilePattern]`, and avoid this clone?

Reply:
(Maybe we even want a version of `is_excluded` that takes a single pattern, since passing a slice here is a little awkward, but not a big deal.)

Reply:
changed `is_excluded` to take an iterator of refs, that way a slice ref can call `iter` and a slice of refs can call `into_iter()`